test/models/fact_check_test.rb

```diff
@@ -741,4 +741,65 @@ def setup
     User.current = nil
     assert_equal u, fc.author
   end
+
+  test "should assign default channel 'manual' for a regular user" do
+    fc = create_fact_check
+    assert_equal 'manual', fc.channel
+  end
+
+  # this was the only way so far I got this test to work so we could set User.current
+  test "should assign default channel 'api' for a BotUser" do
+    bot = create_bot_user(default: true, approved: true)
+    # Prevent duplicate installation errors by stubbing out the default bot callback if needed.
+    Team.any_instance.stubs(:add_default_bots_to_team)
+
+    User.current = bot
+
+    # if we don't skip_check_ability we hit permission errors
+    # if I don't create the claim media, it tries to create a Link Media and hits permissions errors as well
+    media = Media.new(quote: 'blah', skip_check_ability: true)
+    pm = create_project_media(media: media, skip_check_ability: true)
+    cd = create_claim_description(project_media: pm, skip_check_ability: true)
+    fc = create_fact_check(claim_description: cd, channel: nil, skip_check_ability: true)
+
+    assert_equal 'api', fc.channel
+
+    User.current = nil
+  end
+
+  # I feel something like this should work
+  # But if we only rely on User.current this does not work
+  test "should assign default channel 'api' for a BotUser #2" do
```

Comment:
@caiosba for this test to work we need to check `self.user` in set_channel. Will tag you there as well.